### PR TITLE
#1594 request review success toast message

### LINKED ISSUE #1594: Frontend - Request Change Request Reviewer Button should toast success message
### Description

Right now, when you request a reviewer on a change request in Finishline, there's no indication that the request was successfully submitted. Therefore, we should add a toast that pops up with a success message if the update is a success.

### Acceptance Criteria

- [ ] A success toast pops up when a request for review on a change request is successfully submitted
- [ ] The success toast should not show up if the request CR reviewer endpoints errors.

### Proposed Solution

You'll need to use the useToast hook to display the toast. That part of this ticket should be relatively easy, but testing it might be a bit annoying. You'll need to message Reid Chandler on slack to get the slack bot token in order to test the endpoint correctly. 

N.B. you can only request a reviewer on a change request that you've made

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by jakewc12**
## Changes

When a request review success, a toast success pops up confirming that the request was successful.

## Screenshots

image
Screenshot 2023-11-08 at 8 48 14 PM

## Checklist

- [ x] All commits are tagged with the ticket number
- [ x] No linting errors / newline at end of file warnings
- [ x] All code follows repository-configured prettier formatting
- [ x] No merge conflicts
- [ x] All checks passing
- [ x] Screenshots of UI changes (see Screenshots section)
- [ x] Remove any non-applicable sections of this template
- [ x] Assign the PR to yourself
- [ x] No `yarn.lock` changes (unless dependencies have changed)
- [x ] Request reviewers & ping on Slack
- [ x] PR is linked to the ticket (fill in the closes line below)

Closes #1594

### CODE REVIEW

File: src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx [hunk 1]
```diff
@@ -53,6 +53,7 @@ const ChangeRequestActionMenu: React.FC<ChangeRequestActionMenuProps> = ({
     } else {
       try {
         await requestCRReview({ userIds: reviewerIds });
+        toast.success('Request Review successful');
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx:56]: Nice. One change, I think we should change this to say: "Successfully Requested Review From: Thomas Emrax, Joe Schmoe, etc." where you use the userIds to get the users' names from the list of all users in this file

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx [hunk 2]
```diff
@@ -53,6 +53,9 @@ const ChangeRequestActionMenu: React.FC<ChangeRequestActionMenuProps> = ({
     } else {
       try {
         await requestCRReview({ userIds: reviewerIds });
+        reviewerIds.forEach(function (value) {
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx:56]: you should do this like `(userId) => toast.success(...)`, it makes it clearer what variable you're using and arrow functions are a bit cleaner

Comment (RChandler234) [reply src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx:56]: also you should use the userid to retrieve the name of the user of the list of users we have fetched above

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx (hunk 2 above)
Comment (jakewc12) [reply src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx:57]: Got it. Trying that

File: src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx [hunk 3]
```diff
@@ -53,6 +54,9 @@ const ChangeRequestActionMenu: React.FC<ChangeRequestActionMenuProps> = ({
...
+          toast.success('Request Review successful for reviewers ' + getUserFullName(reviewerIds[value]) + value);
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx:58]: we're in the frontend, you can't call this method from the backend, they're two separate processes. You should instead use the list of all users (declared on line 46) and then for each reviewer Id, find the user which has the same id in that list and print their name

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -56,6 +56,7 @@
     "eslint-config-prettier": "^8.5.0",
     "eslint-plugin-prettier": "^4.2.1",
     "prettier": "^2.0.5",
+    "prisma": "latest",
```
Comment (RChandler234) [inline package.json:59]: remove this change to the package.json

File: src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx [hunk 5]
```diff
@@ -52,7 +53,8 @@ const ChangeRequestActionMenu: React.FC<ChangeRequestActionMenuProps> = ({
       toast.error('Must select at least one reviewer to request review from');
     } else {
       try {
-        await requestCRReview({ userIds: reviewerIds });
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx:55]: don't delete this line, it's what's actually sending the request

File: src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx [hunk 6]
```diff
@@ -19,6 +19,7 @@ import { useState } from 'react';
 import ErrorPage from '../ErrorPage';
 import LoadingIndicator from '../../components/LoadingIndicator';
 import { taskUserToAutocompleteOption } from '../../utils/task.utils';
+import { getUserFullName } from '../../../../backend/src/utils/users.utils';
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestActionMenu.tsx:22]: you can't use this method it's from the backend which runs as a separate process from the frontend

**Review by RChandler234 — CHANGES_REQUESTED** [review]
you should untrack the changes to the yarn.lock, you should be able to pull the version from develop, lmk if you have questions on how to do that

**Pull request closed without merging**